pkg/resolver/search.go

```diff
@@ -265,321 +276,69 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
-				// - to be similar to how it is stored in the database
-				if _, ok := arrayProperties[filter.Property]; ok {
-					cleanedVal := make([]string, len(values))
-					for i, val := range values {
-						labels := strings.Split(val, "=")
-						if len(labels) == 2 {
-							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
-						} else if len(labels) == 1 {
-							//// If property is of array type, format it as an array for easy searching
-							cleanedVal[i] = labels[0]
-						} else {
-							klog.Error("Error while decoding label string")
-							cleanedVal[i] = val
+				values := PointerToStringArray(filter.Values)
+
+				if len(propTypeMap) > 0 {
+					if dataTypeInMap, ok := propTypeMap[filter.Property]; !ok { //check if value exists/if value doesn't exist
+						klog.Warningf("Property type for [%s] doesn't exist in cache. Refreshing property type cache",
+							filter.Property)
+						propTypeMapNew, err := GetPropertyTypeCache(ctx) //call database cache again
+						if err != nil {
+							klog.Warningf("Error creating property type map with err: [%s] ", err)
 						}
+						propTypeMap = propTypeMapNew
+						break
 
-					}
-					values = cleanedVal
-				}
+					} else {
 
-				// Check if value is a number or date and get the cleaned up value
-				opDateValueMap := getOperatorAndNumDateFilter(filter.Property, values)
+						klog.V(5).Infof("Prop in map:%s, filter prop is: %s, datatype :%s\n", dataTypeInMap, filter.Property)
+						//if property mactches then call decode function:
+						values, dataTypeFromMap = DecodePropertyTypes(values, dataTypeInMap)
+
+						// Check if value is a number or date and get the cleaned up value
+						opDateValueMap := GetOperatorAndNumDateFilter(filter.Property, values, dataTypeFromMap)
+						//Sort map according to keys - This is for the ease/stability of tests when there are multiple operators
+						keys := GetKeys(opDateValueMap)
+						var operatorWhereDs []exp.Expression //store all the clauses for this filter together
+						for _, operator := range keys {
+							operatorWhereDs = append(operatorWhereDs,
+								GetWhereClauseExpression(filter.Property, operator, opDateValueMap[operator], dataTypeFromMap)...)
+						}
 
-				//Sort map according to keys - This is for the ease/stability of tests when there are multiple operators
-				keys := getKeys(opDateValueMap)
-				var operatorWhereDs []exp.Expression //store all the clauses for this filter together
-				for _, operator := range keys {
-					operatorWhereDs = append(operatorWhereDs,
-						getWhereClauseExpression(filter.Property, operator, opDateValueMap[operator])...)
+						whereDs = append(whereDs, goqu.Or(operatorWhereDs...)) //Join all the clauses with OR
+						fmt.Println(whereDs)
+					}
+				} else {
+					//if map is empty don't return anything
+					klog.Error("Error with property type list is empty.")
+					return nil, nil, err
```

Comment:
If there's any error getting the property type, we should fall back to use the existing logic. This reduces the impact to users in case there's a problem because only a few properties are lists.